translator/navjoy_translator.py

```diff
@@ -0,0 +1,347 @@
+    wzd = wzdx_translator.add_ids(wzd)
+    return wzd
+
+
+def get_linestring_index(map):
+    for i in range(len(map)):
+        if map[i].get("type") == "LineString":
+            return i
+
+
+def get_polygon_index(map):
+    for i in range(len(map)):
+        if map[i].get("type") == "Polygon":
+            return i
+
+
+def get_directions_from_string(directions_string) -> list:
+    if not directions_string or type(directions_string) != str:
+        return []
+
+    directions = []
+
+    # iterate over directions and convert short direction names to WZDx enum directions
+    directions_string = directions_string.strip()
+    for dir in directions_string.split('/'):
+        direction = STRING_DIRECTION_MAP.get(dir.lower())
+        if direction:
+            directions.append(direction)
+
+    return directions
+
+# TODO: Support additional zones per message (streetNameFrom2, ...)
+# Parse Navjoy  to WZDx
+
+
+def parse_reduction_zone(obj, direction, callback_function=None):
+    if not validate_closure(obj):
+        if callback_function:
+            callback_function(obj)
+        return None
+
+    data = obj.get('data', {})
+
+    map = data.get('srzmap')
+    index = get_linestring_index(map)
+    if index != None:
+        coordinates = map[index].get("coordinates", [None])[0]
+    else:
+        index = get_polygon_index(map)
+        if index != None:
+            polygon = map[index].get("coordinates", [None])[0]
+            coordinates = polygon_tools.polygon_to_polyline_center(polygon)
+        else:
+            logging.warning(
+                f"Invalid event with id = {obj.get('sys_gUid')}. No polygon or linestring")
+            return None
+
+    if not coordinates:
+        logging.warning(
+            f"Invalid event with id = {obj.get('sys_gUid')}. No valid coordinates found")
+        return False
+
+    # [0].get('coordinates')[0]
+    # coordinates = polygon_tools.polygon_to_polyline_center(polygon)
+
+    # Reverese polygon if it is in the opposide direction as the message
+    polyline_direction = polygon_tools.get_road_direction_from_coordinates(
+        coordinates)
+    if direction == REVERSED_DIRECTION_MAP.get(polyline_direction):
+        coordinates.reverse()
+
+    geometry = {}
+    if len(coordinates) > 2:
+        geometry['type'] = "LineString"
+    else:
+        geometry['type'] = "MultiPoint"
+
+    geometry['coordinates'] = coordinates
+    properties = wzdx_translator.initialize_feature_properties()
+
+    # Event Type ['work-zone', 'detour']
+    properties['event_type'] = 'work-zone'
+
+    # start_date
+    start_date = date_tools.parse_datetime_from_iso_string(
+        data.get('workStartDate'))
+    properties['start_date'] = date_tools.get_iso_string_from_datetime(
+        start_date)
+
+    # end_date
+    end_date = date_tools.parse_datetime_from_iso_string(
+        data.get('workEndDate'))
+    if end_date:
+        properties['end_date'] = date_tools.get_iso_string_from_datetime(
```

Comment:
Same here - I get that we are checking for a valid end_date value before setting it in the properties object, but could we just use the data.get instead of parsing back to iso string from the datetime?